quickwit/quickwit-search/src/leaf.rs

```diff
@@ -1304,13 +1305,23 @@ pub async fn leaf_search(
         });
     }
 
-    let result = crate::search_thread_pool()
+    let leaf_search_response_result = crate::search_thread_pool()
         .run_cpu_intensive(|| incremental_merge_collector.finalize())
         .instrument(info_span!("incremental_merge_intermediate"))
         .await
-        .context("failed to merge split search responses")??;
+        .context("failed to merge split search responses")?;
```

Comment:
hopefully the collector doesn't panic, but if it does, this should probably count as an error. This is currently not accounted for in metrics

Reply:
Good point. I was sloppy :/ Will fix and merge.